README.md

```diff
@@ -1,14 +1,55 @@
-## How To Write More Correct Software
+# How To Write More Correct Software
 
 Welcome to the How To Write More Correct Software workshop!
-This workshop will help you learn how to use two powerful tools
+
+We're here to help you learn how to use two powerful tools
 to make more of your code more correct:
 
 1. [Dafny](dafny.org), a programming language that formally verifies your implementation matches your specification
 2. [Duvet](https://github.com/awslabs/duvet), a code quality tool for measuring how well your human-readable specification is covered in code
 
 This content is currently a work in progress - use at your own risk, mind our dust, and stay tuned!
 
+## Getting Started
+
+This workshop is best experienced though Visual Studio Code (VS Code) on a sandboxed, pre-built development container.
+There are currently three different options, listed from most to least preferred below.
+
+Once you have your environment, open up [the instructions](./instructions/steps.md) and you're off!
+
+### Codespaces
+
+Click this badge to open this repository in a GitHub Codespace:
+
+[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=542194938)
+
+This is currently only available for users in the [personal account beta](https://docs.github.com/en/codespaces/overview).
+
+### GitPod
+
+Click this button to open this repository in a GitPod workspace:
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/aws-samples/how-to-write-more-correct-software-workshop) 
```

Comment:
This won't work until the PR is merged, but here's the link to open a workspace on the source branch:

https://gitpod.io/#https://github.com/aws-samples/how-to-write-more-correct-software-workshop/tree/add-devcontainer

Reply:
We can speed things up substantially by publishing a pre-built Docker image to https://hub.docker.com/, but I don't want the scope creep of setting that up on this PR.

Reply:
~3 min setup... wow

Reply:
Wow that's slow or wow that's fast? :)

Reply:
Fast